### ISSUE TITLE
manual_is_ascii_check causes error[E0282]: type annotations needed

### ISSUE THREAD

**Issue by m-hugo** [rust-lang/rust-clippy#15326]
### Description

```
The following errors were reported:
error[E0282]: type annotations needed
  --> src/main.rs:82:21
   |
82 |     if "".starts_with(|x| x.is_ascii_digit()){panic!()}
   |                        ^  - type must be known at this point
   |
help: consider giving this closure parameter an explicit type
   |
82 |     if "".starts_with(|x: /* Type */| x.is_ascii_digit()){panic!()}
   |                         ++++++++++++

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0282`.
Original diagnostics will follow.

warning: manual check for common ascii range
  --> src/main.rs:82:24
   |
82 |     if "".starts_with(|x| matches!(x, '0'..='9')){panic!()}
   |                           ^^^^^^^^^^^^^^^^^^^^^^ help: try: `x.is_ascii_digit()`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_is_ascii_check
   = note: `#[warn(clippy::manual_is_ascii_check)]` on by default
```

previously fixed by #11988?

### Version

```text
rustc 1.90.0-nightly (9748d87dc 2025-07-21)
binary: rustc
commit-hash: 9748d87dc70a9a6725c5dbd76ce29d04752b4f90
commit-date: 2025-07-21
host: x86_64-unknown-linux-gnu
release: 1.90.0-nightly
LLVM version: 20.1.8
```
